spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BoxHead.scala

```diff
@@ -0,0 +1,327 @@
+      maxPerImage, numClasses, weight = weight)
+
+    val features = Input()
+    val proposals = Input()
+
+    val boxFeatures = featureExtractor.inputs(features, proposals)
+    val classLogits = clsPredictor.inputs(boxFeatures)
+    val boxRegression = bboxPredictor.inputs(boxFeatures)
+    val result = postProcessor.inputs(classLogits, boxRegression, proposals)
+
+    Graph(Array(features, proposals), Array(boxFeatures, result))
+  }
+
+  private[nn] def clsPredictor(numClass: Int,
+                               inChannels: Int): Module[Float] = {
+    val cls_score = Linear[Float](inChannels, numClass)
+    cls_score.weight.apply1(_ => RNG.normal(0, 0.01).toFloat)
+    cls_score.bias.fill(0.0f)
+    cls_score.asInstanceOf[Module[Float]]
+  }
+
+  private[nn] def bboxPredictor(numClass: Int,
+                               inChannels: Int): Module[Float] = {
+    val bbox_pred = Linear[Float](inChannels, numClass * 4)
+    bbox_pred.weight.apply1(_ => RNG.normal(0, 0.001).toFloat)
+    bbox_pred.bias.fill(0.0f)
+    bbox_pred.asInstanceOf[Module[Float]]
+  }
+
+  private[nn] def featureExtractor(inChannels: Int,
+                                   resolution: Int,
+                                   scales: Array[Float], samplingRatio: Int,
+                                   representationSize: Int): Module[Float] = {
+    val pooler = new Pooler(resolution, scales, samplingRatio)
+    val inputSize = inChannels * math.pow(resolution, 2).toInt
+
+    val fc1 = Linear[Float](inputSize, representationSize, withBias = true)
+      .setInitMethod(Xavier, Zeros)
+    val fc2 = Linear[Float](representationSize, representationSize, withBias = true)
+      .setInitMethod(Xavier, Zeros)
+
+    val model = Sequential[Float]()
+      .add(pooler)
+      .add(InferReshape(Array(0, -1)))
+      .add(fc1)
+      .add(ReLU[Float]())
+      .add(fc2)
+      .add(ReLU[Float]())
+
+    model
+  }
+}
+
+private[nn] class BoxPostProcessor(
+    val scoreThresh: Float,
+    val nmsThresh: Float,
+    val maxPerImage: Int,
+    val nClasses: Int,
+    val weight: Array[Float] = Array(10.0f, 10.0f, 5.0f, 5.0f)
+  ) (implicit ev: TensorNumeric[Float]) extends AbstractModule[Table, Table, Float] {
+
+  private val softMax = SoftMax[Float]()
+  private val nmsTool: Nms = new Nms
+  @transient  private var boxesBuf: Tensor[Float] = null
+
+  /**
+   * Returns bounding-box detection results by thresholding on scores and
+   * applying non-maximum suppression (NMS).
+   */
+  private[nn] def filterResults(boxes: Tensor[Float], scores: Tensor[Float],
+                                numOfClasses: Int): Array[RoiLabel] = {
+    val dim = numOfClasses * 4
+    boxes.resize(Array(boxes.nElement() / dim, dim))
+    scores.resize(Array(scores.nElement() / numOfClasses, numOfClasses))
+
+    val results = new Array[RoiLabel](numOfClasses)
+    var clsInd = 1
+    while (clsInd < numOfClasses) {
+      results(clsInd) = postProcessOneClass(scores, boxes, clsInd)
```

Comment:
where is `results[0]` ?

Reply:
class number 0 is background. So it is ignored. Add some comments for it.